packages/esm-patient-common-lib/src/workspaces/workspaces.ts

```diff
@@ -114,59 +117,75 @@ function getTitleFromExtension(ext: ExtensionRegistration) {
  */
 export function launchPatientWorkspace(name: string, additionalProps?: object) {
   const store = getWorkspaceStore();
-  const state = store.getState();
   const workspace = getWorkspaceRegistration(name);
   const newWorkspace = {
     ...workspace,
     closeWorkspace: (ignoreChanges = true) => closeWorkspace(name, ignoreChanges),
     promptBeforeClosing: (testFcn) => promptBeforeClosing(name, testFcn),
     additionalProps,
   };
-  const existingWorkspaces = state.openWorkspaces.filter((w) => w.type == newWorkspace.type);
 
-  if (existingWorkspaces.length == 0) {
-    store.setState({ ...state, openWorkspaces: [newWorkspace, ...state.openWorkspaces] });
-  } else {
-    const existingIdx = state.openWorkspaces.findIndex((w) => w.name == name);
-    const promptCheckFcn = getPromptBeforeClosingFcn(existingWorkspaces[0].name);
-    if (existingIdx >= 0) {
-      const restOfWorkspaces = [...state.openWorkspaces];
-      restOfWorkspaces.splice(existingIdx, 1);
-      state.openWorkspaces[existingIdx].additionalProps = newWorkspace.additionalProps;
-      const openWorkspaces = [state.openWorkspaces[existingIdx], ...restOfWorkspaces];
-      store.setState({ ...state, openWorkspaces });
-    } else if (!promptCheckFcn || promptCheckFcn()) {
-      const currentName = existingWorkspaces[0].title ?? existingWorkspaces[0].name;
+  const updateStoreWithNewWorkspace = (workspaceToBeAdded: OpenWorkspace, restWorkspaces = null) => {
+    if (!restWorkspaces) {
+      restWorkspaces = store.getState().openWorkspaces;
+    }
+    store.setState({ ...store.getState(), openWorkspaces: [workspaceToBeAdded, ...restWorkspaces] });
+  };
+
+  function promptBeforeLaunchingWorkspace(workspace: OpenWorkspace) {
```

Comment:
I think it would be better to not define this function inside the scope of the `launchPatientWorkspace()` function, especially because it will  call `launchPatientWorkspace()` inside this function.

Reply:
Ohh okay, I understand, I'll make the change!

Reply:
Done!